### PR TITLE
Rework ImputSex to static object

### PR DESCRIPTION
**Pull request by tpoterba**
Add option to use reference population frequency -- more accurate and avoid reading genotypes twice
Use annotations as intermediate allele frequency storage

### CODE REVIEW

**Comment by tpoterba** [comment]
jenkins

File: src/main/scala/org/broadinstitute/hail/methods/ImputeSexPlink.scala [hunk 1]
```diff
@@ -1,129 +1,77 @@
...
+      .mapAnnotations((v: Variant, va: Annotation, gs: Iterable[Genotype]) =>
+        query.map(_.apply(va).orNull)
+          .getOrElse({
+            var nAlt = 0
+            var nTot = 0
+            for (g <- gs) {
+              g.gt.foreach { call =>
+                nAlt += call
+                nTot += 2
+              }
```
Comment (jigold) [inline src/main/scala/org/broadinstitute/hail/methods/ImputeSexPlink.scala]: Does this work in general? I thought g.gt could be anything in the range of 0 to nGenotypes.

**Comment by jigold** [comment]
This looks great Tim! I had one comment/question, but otherwise I think this is ready for Cotton to look at and push into master.

File: src/main/scala/org/broadinstitute/hail/methods/ImputeSexPlink.scala (hunk 1 above)
Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/methods/ImputeSexPlink.scala]: Nice catch, should be g.nNonRefAlleles right?

**Comment by tpoterba** [comment]
I also changed the result to a boolean `isFemale`.  Feel free to push back on this

**Comment by jigold** [comment]
I think the isFemale makes sense if users are storing the reported sex as a Boolean because the next thing someone is going to do is 'sa.sexcheck = sa.imputeSex.isFemale == sa.reportedSex'. The PLINK module does this automatically (1 = False, 2 = True), so the comparison is easy. Not sure how easy it is to compare the result if the reported sex is not a boolean. 'sa.sexcheck = (sa.imputeSex.isFemale && sa.reportedSex == "F") || (sa.imputeSex.isMale && sa.reportedSex == "M")'. I guess that's not terrible to write out, so we should keep isFemale to be consistent with the Fam file import.

**Pull request merged**